osgar/drivers/tankmaria.py

```diff
@@ -0,0 +1,149 @@
+"""
+  Driver for a small tank Maria
+"""
+import math
+
+from osgar.node import Node
+from osgar.bus import BusShutdownException
+
+
+ENC_SCALE = 0.00052
+WHEEL_DISTANCE = 0.4  # 50cm long, 40cm wide
+
+
+class MariaTimer(Node):
+    def __init__(self, config, bus):
+        super().__init__(config, bus)
+        bus.register('tick')
+        self.sleep_time = config['sleep']
+
+    def run(self):
+        try:
+            while self.is_alive():
+                self.publish('tick', None)
+                self.sleep(self.sleep_time)
+
+        except BusShutdownException:
+            pass
+
+
+class RobotTankMaria(Node):
+    def __init__(self, config, bus):
+        super().__init__(config, bus)
+        bus.register('pose2d', 'emergency_stop', 'encoders', 'raw')
+
+        # commands
+        self.desired_speed = 0.0  # m/s
+        self.desired_angular_speed = 0.0
+
+        # status
+        self.emergency_stop = None  # uknown state
+        self.pose = (0.0, 0.0, 0.0)  # x, y in meters, heading in radians (not corrected to 2PI)
```

Comment:
No 3D?

Reply:
there is no way how to get 3D from odometry, right?? It is typically available after fusion with IMU, so yes "no 3D" is expected here.

Reply:
I had no idea if IMU is or isn't available. I didn't even understand the protocol just from reading the code.

In that case, we usually name the beast `pose2d`.

Reply:
OK, fixed